Fix incorrect hostname quoting on Red Hat family OS

ixs opened the pull request:
network.mod_hostname() will incorrectly set a duplicate
hostname string in the configuration file. Fix that.

Fixes #58956

ixs committed:
Fix incorrect hostname quoting on Red Hat family OS

network.mod_hostname() will incorrectly set a duplicate
hostname string in the configuration file. Fix that.

Fixes #58956

Ch3LL reviewed (changes requested):
can you add a test for the issue this fixes?

ixs committed:
Add test for correctly written network config file on RH distros.

Ensure that the hostname written to /etc/sysconfig/networking is
correctly written out, including quoting.

The new tests error out nicely on the previously broken code:

```
E           AssertionError: assert 'HOSTNAME=hostname"hostname\n' == 'HOSTNAME="hostname"\n'
E             - HOSTNAME="hostname"
E             ?                   -
E             + HOSTNAME=hostname"hostname
E             ?          ++++++++

tests/unit/modules/test_network.py:306: AssertionError

E           AssertionError: assert 'HOSTNAME=hostnamehostname\n' == 'HOSTNAME=hostname\n'
E             - HOSTNAME=hostname
E             + HOSTNAME=hostnamehostname
E             ?                  ++++++++

tests/unit/modules/test_network.py:337: AssertionError
```

Passes on the new code.

ixs commented:
@Ch3LL Tests have been added.

Ch3LL reviewed (changes requested):
Can you add a changelog and this will be good to go

ixs committed:
Add changelog entry

ixs commented:
Done. @Ch3LL thanks for the review.
